rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeForFunctionLikeWithinCallLikeArgDeclarationRector.php

```diff
@@ -0,0 +1,221 @@
+
+    public function __construct(
+        private readonly TypeComparator $typeComparator,
+        private readonly PhpVersionProvider $phpVersionProvider,
+        private readonly StaticTypeMapper $staticTypeMapper,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add param types where needed', [
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+(new SomeClass)->process(function ($parameter) {});
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+(new SomeClass)->process(function (string $parameter) {});
+CODE_SAMPLE
+                ,
+                [
+                    new AddParamTypeForFunctionLikeWithinCallLikeArgDeclaration(
+                        'SomeClass',
+                        'process',
+                        0,
+                        0,
+                        new StringType()
+                    ),
+                ]
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param CallLike $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $this->hasChanged = false;
+        foreach ($this->addParamTypeForFunctionLikeParamDeclarations as $addParamTypeForFunctionLikeParamDeclaration) {
+            $type = match (true) {
+                $node instanceof MethodCall => $node->var,
+                $node instanceof StaticCall => $node->class,
+                default => null,
+            };
+
+            if ($type === null) {
+                continue;
+            }
+
+            if (! $this->isObjectType($type, $addParamTypeForFunctionLikeParamDeclaration->getObjectType())) {
+                continue;
+            }
+
+            if (($node->name ?? null) === null) {
+                continue;
+            }
+
+            if (! $node->name instanceof Identifier) {
+                continue;
+            }
+
+            if (! $this->isName($node->name, $addParamTypeForFunctionLikeParamDeclaration->getMethodName())) {
+                continue;
+            }
+
+            $this->processFunctionLike($node, $addParamTypeForFunctionLikeParamDeclaration);
+        }
+
+        if (! $this->hasChanged) {
+            return null;
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param mixed[] $configuration
+     */
+    public function configure(array $configuration): void
+    {
+        Assert::allIsAOf($configuration, AddParamTypeForFunctionLikeWithinCallLikeArgDeclaration::class);
+
+        $this->addParamTypeForFunctionLikeParamDeclarations = $configuration;
+    }
+
+    private function processFunctionLike(
+        CallLike $callLike,
+        AddParamTypeForFunctionLikeWithinCallLikeArgDeclaration $addParamTypeForFunctionLikeWithinCallLikeArgDeclaration
+    ): void {
```

Comment:
first class callable can be checked early here:

```suggestion
    ): void {
    if ($callLike->isFirstClassCallable()) {
            return [];
        }
```